src/control/cmd/daos/container.go

```diff
@@ -113,7 +113,12 @@ func (cmd *containerBaseCmd) openContainer(openFlags C.uint) error {
 
 func (cmd *containerBaseCmd) closeContainer() error {
 	cmd.log.Debugf("closing container: %s", cmd.contUUID)
-	return daosError(C.daos_cont_close(cmd.cContHandle, nil))
+
+	rc := C.daos_cont_close(cmd.cContHandle, nil)
+	if rc == -C.DER_NOMEM {
+		rc = C.daos_cont_close(cmd.cContHandle, nil)
+	}
+	return daosError(rc)
```

Comment:
From a philosophical standpoint, it doesn't seem right that we're adding production code to make the tests work. I understand that this is necessary to handle an injected fault, but it still seems strange to me that we'd handle an OOM situation by retrying the operation. I don't immediately see a better solution, though.

Reply:
In general yes, although OOM is one of the few operations that might work on retry.  Luckily this kind of hack isn't needed in many locations, and the code coverage that it gives us is useful.

Reply:
I would just copy exact wrt disconnectPool (including similar comment) and log error about failed close retry. The function closeContainer should not return an error but should print to ERROR on failure like disconnectPool does.
There's nothing you can do on closeContainer fail but the error should be logged and not returned.
The current closeContainer() calls are ignoring the returned error anyway which is bad practice (e.g. line 97).